deepchecks/nlp/utils/text_properties.py

```diff
@@ -715,79 +772,74 @@ def calculate_builtin_properties(
     english_properties_names = set(ENGLISH_ONLY_PROPERTIES)
-    text_properties_names = {it['name'] for it in text_properties}
-    samples_language = None
-    english_samples = []
-    english_samples_mask = []
-    calculated_properties = {}
-
-    if english_properties_names & text_properties_names:
-        samples_language = run_available_kwargs(
-            language,
-            raw_text=raw_text,
-            **kwargs
-        )
-
-        for lang, text in zip(samples_language, raw_text):
-            if lang == 'en':
-                english_samples.append(text)
-                english_samples_mask.append(True)
-            else:
-                english_samples_mask.append(False)
-
-        new_text_properties = []
-
-        for prop in text_properties:
-            if prop['name'] == 'Language':
-                calculated_properties['Language'] = samples_language
-            else:
-                new_text_properties.append(prop)
-
-        text_properties = new_text_properties
+    text_properties_names = [it['name'] for it in text_properties]
+    calculated_properties = {k: [] for k in text_properties_names}
+
+    # Prepare kwargs for properties that require outside resources:
+    if 'fasttext_model' not in kwargs:
+        kwargs['fasttext_model'] = _get_fasttext_model(models_storage=models_storage)
+
+    if 'cmudict_dict' not in kwargs:
+        properties_requiring_cmudict = list(set(CMUDICT_PROPERTIES) & set(text_properties_names))
+        if properties_requiring_cmudict:
+            if not nltk_download('cmudict', quiet=True):
+                _warn_if_missing_nltk_dependencies('cmudict', format_list(properties_requiring_cmudict))
+                for prop in properties_requiring_cmudict:
+                    calculated_properties[prop] = [np.nan] * len(raw_text)
+            cmudict_dict = corpus.cmudict.dict()
+            kwargs['cmudict_dict'] = cmudict_dict
+
+    is_language_property_requested = 'Language' in [prop['name'] for prop in text_properties]
+    # Remove language property from the list of properties to calculate as it will be calculated separately:
+    if is_language_property_requested:
+        text_properties = [prop for prop in text_properties if prop['name'] != 'Language']
 
     warning_message = (
         'Failed to calculate property {0}. '
         'Dependencies required by property are not installed. '
         'Error:\n{1}'
     )
+    import_warnings = set()
 
     progress_bar = create_progress_bar(
-        iterable=list(text_properties),
-        name='Text Properties Calculation',
-        unit='Text Property'
+        iterable=list(raw_text),
+        name='Text Samples Calculation',
+        unit='Text Sample'
     )
-
-    # TODO: refactor
-    for prop in progress_bar:
+    for text in progress_bar:
         progress_bar.set_postfix(
-            {'Property': prop['name']},
+            {'Sample': truncate_string(text, max_length=20) if text else 'EMPTY STRING'},
```

Comment:
make EMPTY STRING a const var

Reply:
Why though? Just used here and for display purposes...

Reply:
Just so its clearer to find in case later we would wanna change it